src/vivarium_sodium_reduction/components/risks.py

```diff
@@ -0,0 +1,168 @@
+from vivarium_public_health.utilities import EntityString
+
+class DropValueRisk(Risk):
+    def __init__(self, risk: str):
+        super().__init__(risk)
+        self.raw_exposure_pipeline_name = f"{self.risk.name}.raw_exposure"
+        self.drop_value_pipeline_name = f"{self.risk.name}.drop_value"
+
+    def setup(self, builder: Builder) -> None:
+        super().setup(builder)
+        self.raw_exposure = self.get_raw_exposure_pipeline(builder)
+        self.drop_value = self.get_drop_value_pipeline(builder)
+
+    def get_drop_value_pipeline(self, builder: Builder) -> Pipeline:
+        return builder.value.register_value_producer(
+            self.drop_value_pipeline_name,
+            source=lambda index: pd.Series(0.0, index=index),
+        )
+
+    def get_raw_exposure_pipeline(self, builder: Builder) -> Pipeline:
+        return builder.value.register_value_producer(
+            self.raw_exposure_pipeline_name,
+            source=self.get_current_exposure,
+            requires_columns=["age", "sex"],
+            requires_values=[self.propensity_pipeline_name],
+        )
+
+    def get_exposure_pipeline(self, builder: Builder) -> Pipeline:
+        return builder.value.register_value_producer(
+            self.exposure_pipeline_name,
+            source=self.get_current_exposure,
+            requires_columns=["age", "sex"],
+            requires_values=[self.propensity_pipeline_name],
+            preferred_post_processor=self.get_drop_value_post_processor(builder, self.risk),
+        )
+
+    def get_drop_value_post_processor(self, builder: Builder, risk: EntityString):
+        drop_value_pipeline = builder.value.get_value(self.drop_value_pipeline_name)
+
+        def post_processor(exposure, _):
+            drop_values = drop_value_pipeline(exposure.index)
+            return exposure - drop_values
+
+        return post_processor
+
+class CorrelatedRisk(DropValueRisk):
+    @property
+    def columns_created(self) -> List[str]:
+        return []
+
+    @property
+    def columns_required(self) -> Optional[List[str]]:
+        return [self.propensity_column_name]
+
+    @property
+    def initialization_requirements(self) -> Dict[str, List[str]]:
+        return {
+            "requires_columns": [],
+            "requires_values": [],
+            "requires_streams": [],
+        }
+
+    def on_initialize_simulants(self, pop_data: SimulantData) -> None:
+        pass
+
+    def on_time_step_prepare(self, event: Event) -> None:
+        pass
+
+class RiskCorrelation(Component):
+    @property
+    def columns_created(self) -> List[str]:
+        return self.propensity_column_names + self.exposure_column_names
+
+    @property
+    def columns_required(self) -> Optional[List[str]]:
+        return ["age"]
+
+    @property
+    def initialization_requirements(self) -> Dict[str, List[str]]:
+        return {"requires_columns": ["age"]}
```

Comment:
Line 93 should be updated to `return {"requires_columns": ["age"] + self.ensemble_propensities}`